CMakeLists.txt

```diff
@@ -5,122 +5,202 @@ set(CMAKE_CXX_STANDARD_REQUIRED on)
-link_directories(${ICU_LIBRARY_DIRS})
-
 set(EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR}/bin)
 set(LIBRARY_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR}/lib)
-#link_directories(${LIBRARY_OUTPUT_PATH})
-#cmake_policy(CMP0015 NEW)
 
 list(APPEND CMAKE_MODULE_PATH
-	"${CMAKE_CURRENT_SOURCE_DIR}/lib/liblcf/builds/cmake/Modules"
 	"${CMAKE_CURRENT_SOURCE_DIR}/builds/cmake/Modules")
 
-add_definitions(
-  ${PNG_DEFINITIONS}
-
-  -D USE_SDL=1)
+if(CMAKE_BUILD_TYPE MATCHES "Debug")
+	add_definitions(-D_DEBUG=1)
+endif()
+	
+add_definitions(-DUSE_SDL=1)
 
 if(CMAKE_GENERATOR MATCHES "Makefile")
-  add_definitions(-fno-rtti)
+	add_definitions(-fno-rtti)
 endif()
 
-# wine checking
-find_program(WINE_EXECUTABLE wine)
-if(EXISTS ${WINE_EXECUTABLE})
-  message(STATUS "wine registry enabled")
-  add_definitions(-DHAVE_WINE=1)
+# Endianess check
+if(NOT DEFINED CMAKE_TOOLCHAIN_FILE)
+	include(TestBigEndian)
+	test_big_endian(WORDS_BIGENDIAN)
+	if(WORDS_BIGENDIAN)
+		add_definitions(-DWORDS_BIGENDIAN=1)
+	endif()
 endif()
 
-# shinonome font
-set(SHINONOME_PATH "${CMAKE_CURRENT_SOURCE_DIR}/resources/shinonome")
-set(SHINONOME_SRC_PATH "${CMAKE_CURRENT_SOURCE_DIR}/src")
-set(SHINONOME_SRCS "${SHINONOME_SRC_PATH}/shinonome_gothic.cpp"
-  "${SHINONOME_SRC_PATH}/shinonome_mincho.cpp"
-  "${SHINONOME_SRC_PATH}/shinonome.h")
-find_package(Ruby)
-if(RUBY_FOUND)
-  set(SHINONOME_GENERATOR "${SHINONOME_PATH}/generate_cxx_font.rb")
-  add_custom_command(
-    OUTPUT ${SHINONOME_SRCS}
-    COMMAND "${RUBY_EXECUTABLE}" ARGS "${SHINONOME_GENERATOR}"
-    DEPENDS "${SHINONOME_GENERATOR}"
-    WORKING_DIRECTORY "${SHINONOME_PATH}")
-endif()
+# Adds a library to the liblist if COND is true and the lib is found
+# When REQ is true the lib is required
+function(find_lib_int LIBNAME COND REQ)
+	if(${COND})
+		list(APPEND EASYRPG_LIB_LIST ${LIBNAME})
+	
+		if(${REQ})
+			find_package(${LIBNAME} REQUIRED)
+		else()
+			find_package(${LIBNAME})
+		endif()
+		
+		string(TOUPPER ${LIBNAME} LIBNAMEUP)
+		include_directories(${${LIBNAMEUP}_INCLUDE_DIR} ${${LIBNAMEUP}_INCLUDE_DIRS})
+		set(NEW_LIBS ${${LIBNAMEUP}_LIBRARY} ${${LIBNAMEUP}_LIBRARIES} PARENT_SCOPE)
+	endif()
+
+	set(RETVAL ${${LIBNAME}_FOUND} PARENT_SCOPE)
+	# Export _FOUND variable to parent scope
+	set(${LIBNAME}_FOUND ${${LIBNAME}_FOUND} PARENT_SCOPE)
+endfunction()
+
+function(find_lib_req LIBNAME COND)
+	find_lib_int(${LIBNAME} ${COND} ON)
+	
+	# Export _FOUND variable to parent scope which is global scope
+	set(${LIBNAME}_FOUND ${${LIBNAME}_FOUND} PARENT_SCOPE)
+	set(EASYRPG_PLAYER_LIBRARIES ${EASYRPG_PLAYER_LIBRARIES} ${NEW_LIBS} PARENT_SCOPE)
+	set(RETVAL ${RETVAL} PARENT_SCOPE)
+endfunction()
+
+function(find_lib LIBNAME COND)
+	find_lib_int(${LIBNAME} ${COND} OFF)
+	
+	# Export _FOUND variable to parent scope which is global scope
+	set(${LIBNAME}_FOUND ${${LIBNAME}_FOUND} PARENT_SCOPE)
+	set(EASYRPG_PLAYER_LIBRARIES ${EASYRPG_PLAYER_LIBRARIES} ${NEW_LIBS} PARENT_SCOPE)
+	set(RETVAL ${RETVAL} PARENT_SCOPE)
```

Comment:
Yes is just my pseudo-return variable because looks like that you can't really return something from Cmake functions...